src/inc/Admin/EmailLogsTab.php

```diff
@@ -121,17 +122,61 @@ public function process_email_content_preview() {
             return;
         }
 
+        $allowed_capability = WPML_Init::getInstance()->getService( 'plugin' )->getSetting( 'can-see-submission-data', 'manage_options' );
+
+        if ( ! current_user_can( $allowed_capability ) ) {
+            return;
+        }
+
         // Get the mail log.
         $mail = Mail::find_one( absint( $_GET['email_log_id'] ) );
 
         if ( ! $mail ) {
             return;
         }
 
-        echo $mail->get_message();
+        echo $this->get_html_preview_message( $mail->get_message() );
         exit;
     }
 
+    /**
+     * Get the message to be rendered in the preview.
+     *
+     * @since {VERSION}
+     *
+     * @param string $message Email log message.
+     *
+     * @return string
+     */
+    private function get_html_preview_message( $message ) {
+
+        // Loosely test if the message is HTML.
+        if ( preg_match( '/<[a-z][^\s>\/]*>/i', $message ) ) {
+            // Convert escape '<script`> tags.
+            $message = str_replace(
+                [
+                    '<script',
+                    '</script>',
+                ],
+                [
+                    esc_html( '<script' ),
+                    esc_html( '</script>' ),
+                ],
+                $message
+            );
+
+            // Removes `<xml>` tags.
+            $message = preg_replace( '/<xml\b[^>]*>(.*?)<\/xml>/is', '', $message );
+
+            $allowed_html = wp_kses_allowed_html( 'post' );
+            $allowed_html['style'][''] = true;
+
+            return wp_kses( $message, $allowed_html );
```

Comment:
This part will change the way the emails look, so we would like to skip this part.

We are already escaping the script tags and removing xml tags, so the XSS issue should be fixed.